examples/minimal/packages/client-react/src/mud/setup.ts

```diff
@@ -48,11 +48,25 @@ export async function setup() {
     setInterval(requestDrip, 20000);
   }
 
+  const signerOrProvider = signer ?? result.network.providers.get().json;
   // Create a World contract instance
-  const worldContract = IWorld__factory.connect(
-    networkConfig.worldAddress,
-    signer ?? result.network.providers.get().json
-  );
+  const worldContract = IWorld__factory.connect(networkConfig.worldAddress, signerOrProvider);
+
+  const currentBlockNumber = await awaitStreamValue(result.network.blockNumber$);
```

Comment:
maybe we should just ping the RPC for the current block number here to not block for too long in case the chain has slow block times

Reply:
added this